### PR TITLE
fix: add IPython as Sphinx extension

### PR DESCRIPTION
**Pull request by agoose77**
This PR fixes the root cause that motivated #1964. In fact, the problem was exclusively on RTD, but when I checked my original fix, I checked the CI build result (which has not shown this highlighting problem). 

This PR adds IPython as a Sphinx extension, and reverts the changes to the `code-cell` language.

----
:books: The documentation for this PR will be available at <https://awkward-array.readthedocs.io/en/agoose77-docs-fix-syntax-highlighting/> once Read the Docs has finished building :hammer:

### CODE REVIEW

**Review by jpivarski — APPROVED** [review]
I don't see a difference in syntax coloring between

https://awkward-array.readthedocs.io/en/main/reference/generated/ak.cartesian.html

and

https://awkward-array.readthedocs.io/en/agoose77-docs-fix-syntax-highlighting/reference/generated/ak.cartesian.html

but I guess the distinction would be in the `%magic` commands (IPython vs Python).

But this has my approval and can be merged when you're ready.

**Comment by jpivarski** [comment]
The last message was an hour ago, so I guess you're done with this PR. Half the tests didn't run because it's a documentation-only change. I'll press the red button and merge it.

**Comment by agoose77** [comment]
@jpivarski this PR is fixing the user guide, whose code cells are unstyled in the RTD main.

**Pull request merged**
